### PR TITLE
EOS-22429: Added Alex workflow

### PR DESCRIPTION
**Pull request by venkuppu-chn**
Signed-off-by: Venkatesh K <venkatesh.k@seagate.com>
Use of Inclusive words are harmful such as gender favoring, polarizing, race related, religion inconsiderate, or other unequal phrasing in text. Alex is the tool to scan the inclusive words and post the results in PR annotations.

### CODE REVIEW

**Comment by suvratjoshi** [comment]
hi @venkuppu-chn ,
Can you please add description to commit message and PR summary please?

Regarding the changes :
As I understand, this tool addition is for catching the contentious keywords automatically.
I believe we already used to have periodic flexera scans for catching such keywords and generating Jenkins reports for the same.
Are we replacing that flexera scanning with this?

**Comment by venkuppu-chn** [comment]
> hi @venkuppu-chn ,
> Can you please add description to commit message and PR summary please?
> 
> Regarding the changes :
> As I understand, this tool addition is for catching the contentious keywords automatically.
> I believe we already used to have periodic flexera scans for catching such keywords and generating Jenkins reports for the same.
> Are we replacing that flexera scanning with this?

Flexera scan only for security vulnerabilities. It will not scan a any inclusive word (Problematic words) and report it in PR.

**Comment by suvratjoshi** [comment]
> @suvratjoshi Flexera scan only for security vulnerabilities. It will not scan a inclusive word (Problematic words) and report it in PR.
Okay, got it.
But as I remember, we can add additional problem keywords in flexera.
So I am just curious to know, if we already have flexera and still running on a periodic basis for each repo, then why are we not adding these additional problem keywords there, so that flexera will automatically report it, along with the current job it is already performing.

Hi @pujamudaliar, 
Do we still run flexera scans for all repo's periodically?

**Review by suvratjoshi — APPROVED** [review]

**Comment by suvratjoshi** [comment]
@venkuppu-chn ,
Thanks for adding PR description.
Can you confirm if commit message is also added?
Also please make a single commit.
Thanks!

**Comment by mssawant** [comment]
@venkuppu-chn, please post a link to some document or confluence page regarding approval of this tool for its usage in Seagate, if you can post some details (documentation) regarding the decision making about incorporating this tool then it would be good.

File: .github/workflows/alex_reviewdog.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
```
Comment (mssawant) [inline .github/workflows/alex_reviewdog.yml:2]: 2021? as this is a new file.

**Review by mssawant — APPROVED** [review]
Minor comment.

File: .github/workflows/alex_reviewdog.yml (hunk 1 above)
Comment (venkuppu-chn) [reply .github/workflows/alex_reviewdog.yml]: Thanks Mandar for pointing this. I have changed the year to 2021.

**Review by suvratjoshi — APPROVED** [review]

**Review by mssawant — APPROVED** [review]

**Pull request merged**